dimod/discrete/cydiscrete_quadratic_model.pyx

```diff
@@ -435,32 +326,14 @@ cdef class cyDiscreteQuadraticModel:
                     raise ValueError("case {} is invalid, variable only supports {} "
                                      "cases".format(case_v, self.num_cases(v)))
 
-                cu = self.case_starts_[u] + case_u
-                cv = self.case_starts_[v] + case_v
+                self.dqm_.set_quadratic_case(u, case_u, v, case_v, bias)
 
-                self.bqm_.set_quadratic(cu, cv, bias)
+            self.dqm_.connect_variables(u,v)
```

Comment:
@arcondello  I put this line of code only for a test case to pass, I think I should remove it and modify the test case, instead

```
350 dqm.set_quadratic('a', 'b', {})
351
352 self.assertEqual(dqm.get_quadratic('a', 'b'), {})
```

since if we serialize this dqm, and de-serialize differences will arise, as connections among variables are recreated based on actual connections among cases. I can fix it in this PR or in a different PR (since this changes the original behaviour of the code, so far it just imitates)

Reply:
Right, I know I told you the old behavior was a bug, but I think it is in fact meaningful for `dqm.set_quadratic(u, v, {})` to set the interaction, but not biases. I know it's weird, but I suppose I can imagine a case where I populate the structure of the DQM before setting the biases where I would want to do that.